src/dvclive/live.py

```diff
@@ -271,8 +271,8 @@ def log_param(
         """Saves the given parameter value to yaml"""
         self.log_params({name: val})
 
-    def make_summary(self):
-        if self._step is not None:
+    def make_summary(self, update_step: bool = True):
+        if self._step is not None and update_step:
             self.summary["step"] = self.step
         dump_json(self.summary, self.metrics_file, cls=NumpyEncoder)
```

Comment:
Minor: 
I worry about the user interaction because we made `summary` a public property that can be modified. 
We would be breaking existing usage and we could not care much because we didn't publicize. However, we would need to be more careful about documenting this hidden reset and how to properly modify the public summary`.

Major:
TBH I am divided about the relevance or what would be the "right" behavior in #239. 
What would happen in scenarios like the `lightning` logger, which logs `step` and `epoch` metrics at different intervals?

Reply:
My minor concern is that documenting this extra arg feels like it's exposing internals to the user, and calling `make_summary` manually becomes more complicated because users need to understand whether to use `update_step`.

However, overall I see that there are benefits to both approaches, so not a blocker if you prefer the current approach. In either case, we can close #239 as either not planned or resolved.

Reply:
> documenting this extra arg feels like it's exposing internals to the user

I feel like we could not document it at all